### PR TITLE
Implementation of ClearRect with masking support

### PR DESCRIPTION
**Pull request by msft-Jeyaram**

### CODE REVIEW

**Comment by msft-Jeyaram** [comment]
Note: the second iteration is the change from the last CR.
The implementation was correct, it was the way how tests were that caused some failures, which are fixed now.

@DHowett-MSFT please take a look at the new heart art :)

**Comment by msft-Jeyaram** [comment]
Looks like git lfs is failing on build machines

File: Frameworks/CoreGraphics/CGContext.mm [hunk 1]
```diff
@@ -2297,22 +2300,46 @@ void CGContextShowGlyphsWithAdvances(CGContextRef context, const CGGlyph* glyphs
 #pragma endregion
 
 #pragma region Drawing Operations - Basic Shapes
+
+HRESULT __CGContext::ClearRect(CGRect rect) {
+    PushBeginDraw();
+    auto endDraw = wil::ScopeExit([this]() { PopEndDraw(); });
```
Comment (DHowett-MSFT) [inline Frameworks/CoreGraphics/CGContext.mm]: prefer `return PopEndDraw()` at the end; otherwise, we lose the HRESULT from EndDraw.

**Comment by msft-Jeyaram** [comment]
so on the build machine, the generated image for one of the tests differ by a pixel :(
making it such that we check if the pixel is present.

**Comment by msft-Jeyaram** [comment]
Verbal extra sign off by @DHowett-MSFT  merging .

**Pull request merged**